File: packages/sdk/csharp/packages/admin/D1Client.cs
using System;
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;
using EdgeBase.Generated;
// EdgeBase C# Admin SDK — D1 Client
// Cloudflare D1 데이터베이스 Raw SQL 실행. Service Key 전용.
// All HTTP calls delegate to Generated Admin Core (GeneratedAdminApi).
// No hardcoded API paths — the core is the single source of truth.

namespace EdgeBase.Admin
{

/// <summary>D1 데이터베이스 클라이언트.</summary>
public sealed class D1Client
{
    private readonly JbHttpClient _http;
    private readonly GeneratedAdminApi _adminCore;
    private readonly string _database;

    internal D1Client(JbHttpClient http, string database)
    {
        _http      = http;
        _adminCore = new GeneratedAdminApi(http);
        _database  = database;
    }

    /// <summary>SQL 쿼리 실행.</summary>
    /// <param name="query">SQL 쿼리 (? 바인드 변수 사용)</param>
    /// <param name="parameters">바인드 파라미터</param>
    /// <returns>결과 행 목록</returns>
    public async Task<List<Dictionary<string, object?>>> ExecAsync(
        string query, object[]? parameters = null, CancellationToken ct = default)
    {
        var body = new Dictionary<string, object?>
        {
            ["query"]  = query,
            ["params"] = parameters ?? Array.Empty<object>()
        };
        var result = await _adminCore.ExecuteD1QueryAsync(
            Uri.EscapeDataString(_database), body, ct);
        return JsonHelper.ExtractList(result, "results");
    }
}
}

File: packages/sdk/dart/packages/admin/lib/src/d1_client.dart
/// D1Client — D1 database access for server-side use.
///
/// ```dart
/// final rows = await admin.d1('analytics').exec(
///   'SELECT * FROM events WHERE type = ?', ['click'],
/// );
/// ```
import 'package:edgebase_core/src/http_client.dart';
import 'generated/admin_api_core.dart';

/// Client for a user-defined D1 database.
class D1Client {
  final GeneratedAdminApi _core;
  final String _database;

  D1Client(HttpClient http, this._database) : _core = GeneratedAdminApi(http);

  /// Execute a SQL query. Use ? placeholders for bind parameters.
  /// All SQL is allowed (DDL included).
  Future<List<dynamic>> exec(String query, [List<dynamic>? params]) async {
    final body = <String, dynamic>{'query': query};
    if (params != null) body['params'] = params;
    final res = await _core.executeD1Query(_database, body);
    if (res is Map<Object?, Object?> && res['results'] is List) {
      return res['results'] as List;
    }
    return [];
  }

  /// Alias for exec() to match SDK parity across runtimes.
  Future<List<dynamic>> query(String query, [List<dynamic>? params]) => exec(query, params);
}
